### PR TITLE
Move all WCF targets to NLog.Wcf 

### LINKED ISSUE #2582: Move WCF targets to NLog.WCF package
and move to NLog.WCF

See also https://github.com/NLog/NLog/issues/1955

### PR DESCRIPTION
**Pull request by snakefoot**
Also:

- Removed System.ServiceModel as dependency for NLog, see #1955
- Improved error handling in LogReceiverWebServiceTarget

For NLog 5.0 (Instead use NLog.Wcf). Resolves #2582

---
This change is [Reviewable](https://reviewable.io/reviews/nlog/nlog/2658)

### CODE REVIEW

File: src/NLog/LogReceiverService/SoapLogReceiverClient.cs [hunk 1]
```diff
@@ -1,98 +0,0 @@
...
-    using System.Web.Services;
-    using System.Web.Services.Description;
-    using System.Web.Services.Protocols;
-    using System.Xml.Serialization;
-
-    /// <summary>
-    /// Log Receiver Client using legacy SOAP client.
-    /// </summary>
-    [WebServiceBindingAttribute(Name = "BasicHttpBinding_ILogReceiverServer", Namespace = "http://tempuri.org/")]
-    public class SoapLogReceiverClient : SoapHttpClientProtocol, ILogReceiverClient
```
Comment (304NotModified) [inline src/NLog/LogReceiverService/SoapLogReceiverClient.cs]: Why is this one removed?

Comment (snakefoot) [reply src/NLog/LogReceiverService/SoapLogReceiverClient.cs]: It is legacy from from when NLog supported the Net20-platform (and earlier) without WCF-support.

**Comment by 304NotModified** [comment]
@snakefoot FYI, I think I will start with merging the 5.0 PRs into dev this week.

(just got a random PR to announce this ;))

master will be for 4.6.3 as we are doing [gitflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow)

**Comment by snakefoot** [comment]
> FYI, I think I will start with merging the 5.0 PRs into dev this week.

That sounds good. Hopefully there will not be more latenight-blunders from my side in NLog 4.6 (but never say never).

**Comment by snakefoot** [comment]
Think I have killed the Silverlight support for pre-WCF-logic. But I think that is perfectly okay for NLog 5.0

**Comment by snakefoot** [comment]
I guess a bonus PR would be to move the source-code files into the NLog.Wcf folder. And completely remove `WCF_SUPPORTED`

**Comment by 304NotModified** [comment]
Good idea, could you please create an issue for that?

**Comment by snakefoot** [comment]
Created #3367

**Pull request merged**